services/dynamic/dynamic-yaml.service.js

```diff
@@ -6,29 +6,53 @@ import jsonPath from './json-path.js'
 export default class DynamicYaml extends jsonPath(BaseYamlService) {
   static enabledMetrics = [MetricNames.SERVICE_RESPONSE_SIZE]
   static route = createRoute('yaml')
-  static examples = [
-    {
-      title: 'Dynamic YAML Badge',
-      namedParams: {},
-      queryParams: {
-        url: 'https://raw.githubusercontent.com/badges/shields/master/.github/dependabot.yml',
-        query: '$.version',
-        prefix: '[',
-        suffix: ']',
-      },
-      documentation: `<p>
-        The dynamic YAML badge takes two required query params: <code>url</code> and <code>query</code>.
-        <ul>
-          <li><code>url</code> is the URL to a YAML document</li>
-          <li><code>query</code> is a <a href="https://jsonpath.com/">JSONPath</a> expression that will be used to query the document</li>
-        </ul>
-        Also an optional <code>prefix</code> and <code>suffix</code> may be supplied.
-      </p>`,
-      staticPreview: {
-        message: '2',
+  static openApi = {
+    '/badge/dynamic/yaml': {
```

Comment:
Probably not worth it, at least not as part of this PR, but intuition is that we should either try to extract this from the `route` variable that's assigned a value above at the `createRoute` call site, or perhaps we should just make the route explicit inline here; It just gives me pause that we've got a hardcoded value here two lines down from where this is set dynamically in a function call, even though I know we're not changing the actual route values ever

Reply:
Agreed. This comes a few steps further on in the migration plan I laid out.
This is exactly the sort of thing I have in mind when I say "look for ways we can reduce the duplication between the service definition and OpenAPI spec" in the top post - there is definitely information we are specifying twice.
I think I want to have a slightly more holistic view of it all first to arrive at the right abstractions.